packages/smooth_app/lib/cache/files/files_cache_impl.dart

```diff
@@ -0,0 +1,129 @@
+// ignore_for_file: avoid_slow_async_io
+part of 'files_cache.dart';
+
+/// Implementation based on [getTemporaryDirectory]
+class _ShortLivingFileCacheManagerImpl extends _FileCacheManagerImpl {
+  _ShortLivingFileCacheManagerImpl({
+    required String subFolderName,
+  }) : super._(
+          rootDirectory: getTemporaryDirectory(),
+          subFolderName: subFolderName,
+        );
+}
+
+/// Implementation based on [getApplicationDocumentsDirectory]
+class _LongLivingFilesCacheManagerImpl extends _FileCacheManagerImpl {
+  _LongLivingFilesCacheManagerImpl({
+    required String subFolderName,
+  }) : super._(
+          rootDirectory: getApplicationDocumentsDirectory(),
+          subFolderName: subFolderName,
+        );
+}
+
+class _FileCacheManagerImpl extends FileCache {
```

Comment:
This class is very good by itself. Could be reused in different projects.
My suggestion: put it in a distinct file named `file_cache_manager_impl.dart` and name the class `FileCacheManagerImpl`.